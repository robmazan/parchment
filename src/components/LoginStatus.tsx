import React from "react";
import { User } from "../slices/userSlice";
import { connect } from "react-redux";
import { getUserFullName, getIsLoggedIn } from "../slices/userSlice";

type OwnProps = {
  name?: string;
  isLoggedIn: boolean;
  loginURI: string;
  logoutURI: string;
};

type ComponentSettings = {
  linkURI: string;
  linkText: string;
  linkClass: string;
  containerClass: string;
  nameComponent?: JSX.Element;
};

const getSettings = (props: OwnProps): ComponentSettings => {
  const { isLoggedIn, logoutURI, loginURI, name } = props;
  if (isLoggedIn) {
    return {
      linkURI: logoutURI,
      linkText: "Logout",
      linkClass: "login-status__link--small",
      containerClass: "login-status--small",
      nameComponent: <div className="login-status__name">{name}</div>
    };
  } else {
    return {
      linkURI: loginURI,
      linkText: "Login",
      linkClass: "login-status__link",
      containerClass: "login-status"
    };
  }
};

export const LoginStatus: React.FC<OwnProps> = props => {
  const {
    containerClass,
    nameComponent,
    linkClass,
    linkURI,
    linkText
  } = getSettings(props);

  return (
    <div className={containerClass}>
      {nameComponent}
      <a className={linkClass} href={linkURI}>
        {linkText}
      </a>
    </div>
  );
};

type StateProps = {
  user: User;
};

/* istanbul ignore next */
const mapStateToProps = (state: StateProps) => ({
  isLoggedIn: getIsLoggedIn(state.user),
  name: getUserFullName(state.user)
});

export default connect(mapStateToProps)(LoginStatus);
