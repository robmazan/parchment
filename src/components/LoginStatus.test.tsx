import React from "react";
import { shallow } from "enzyme";
import { LoginStatus } from "./LoginStatus";

describe("LoginStatus", () => {
  it("renders login link for not logged in user", () => {
    const component = shallow(
      <LoginStatus
        loginURI="/login"
        logoutURI="/logout"
        isLoggedIn={false}
        name=""
      />
    );
    expect(component).toMatchSnapshot();
  });

  it("renders logout link for logged in user", () => {
    const component = shallow(
      <LoginStatus
        loginURI="/login"
        logoutURI="/logout"
        isLoggedIn={true}
        name="Test User"
      />
    );
    expect(component).toMatchSnapshot();
  });
});
