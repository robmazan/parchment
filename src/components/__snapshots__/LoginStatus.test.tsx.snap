// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginStatus renders login link for not logged in user 1`] = `
<div
  className="login-status"
>
  <a
    className="login-status__link"
    href="/login"
  >
    Login
  </a>
</div>
`;

exports[`LoginStatus renders logout link for logged in user 1`] = `
<div
  className="login-status--small"
>
  <div
    className="login-status__name"
  >
    Test User
  </div>
  <a
    className="login-status__link--small"
    href="/logout"
  >
    Logout
  </a>
</div>
`;
